wicket/src/wicketd.rs

```diff
@@ -0,0 +1,130 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Code for talking to wicketd
+
+use crate::Event;
+use slog::{o, warn, Logger};
+use std::net::SocketAddrV6;
+use std::sync::mpsc::Sender;
+use tokio::sync::mpsc;
+use tokio::time::{interval, Duration, MissedTickBehavior};
+use wicketd_client::types::RackV1Inventory;
+
+const WICKETD_POLL_INTERVAL: Duration = Duration::from_secs(5);
+const WICKETD_TIMEOUT_MS: u32 = 1000; // 3 sec
+
+// Assume that these requests are periodic on the order of seconds or the
+// result of human interaction. In either case, this buffer should be plenty
+// large.
+const CHANNEL_CAPACITY: usize = 1000;
```

Comment:
This is probably a question for a later PR (I don't think this is actually used yet, right?), but:

a) What happens if the channel fills completely?
b) What's the effect if the channel isn't completely full but has several hundred items in it?

If the answer to (a) is "the UI becomes unresponsive" and the answer to (b) is "the UI remains responsive but is very out of date", I'm not sure that's a whole lot better. Maybe a smaller channel with a way to politely block the UI or give the user a way to cancel outstanding requests? That sounds pretty complicated already, so maybe disregard. Just wanted to dump some thoughts while I was reading.

Reply:
These are great questions. As you noted the channel isn't in use yet, but will be soon. I opened an issue to track: https://github.com/oxidecomputer/omicron/issues/1942

Reply:
We've been using wicket for about a year now and have not seen an issue here. Closing this.